overlord/devicestate/devicestate_bootconfig_test.go

```diff
@@ -0,0 +1,233 @@
+	"github.com/snapcore/snapd/overlord/auth"
+	"github.com/snapcore/snapd/overlord/devicestate"
+	"github.com/snapcore/snapd/overlord/devicestate/devicestatetest"
+	"github.com/snapcore/snapd/overlord/snapstate"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/snap"
+)
+
+type deviceMgrBootconfigSuite struct {
+	deviceMgrBaseSuite
+
+	managedbl *bootloadertest.MockTrustedAssetsBootloader
+}
+
+var _ = Suite(&deviceMgrBootconfigSuite{})
+
+func (s *deviceMgrBootconfigSuite) SetUpTest(c *C) {
+	s.deviceMgrBaseSuite.SetUpTest(c)
+
+	s.managedbl = bootloadertest.Mock("mock", c.MkDir()).WithTrustedAssets()
+	bootloader.Force(s.managedbl)
+
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	devicestate.SetBootOkRan(s.mgr, true)
+	si := &snap.SideInfo{
+		RealName: "pc",
+		Revision: snap.R(33),
+		SnapID:   "foo-id",
+	}
+	snapstate.Set(s.state, "pc", &snapstate.SnapState{
+		SnapType: "gadget",
+		Sequence: []*snap.SideInfo{si},
+		Current:  si.Revision,
+		Active:   true,
+	})
+	s.state.Set("seeded", true)
+
+	// minimal mocking to reach the mocked bootloader API call
+	modeenv := boot.Modeenv{
+		Mode:           "run",
+		RecoverySystem: "",
+		CurrentKernelCommandLines: []string{
+			"snapd_recovery_mode=run console=ttyS0 console=tty1 panic=-1",
+		},
+	}
+	err := modeenv.WriteTo("")
+	c.Assert(err, IsNil)
+}
+
+func (s *deviceMgrBootconfigSuite) setupUC20Model(c *C) *asserts.Model {
+	devicestatetest.SetDevice(s.state, &auth.DeviceState{
+		Brand:  "canonical",
+		Model:  "pc-model-20",
+		Serial: "didididi",
+	})
+	return s.makeModelAssertionInState(c, "canonical", "pc-model-20", mockCore20ModelHeaders)
+}
+
+func (s *deviceMgrBootconfigSuite) testBootConfigUpdateRun(c *C, updateAttempted, applied bool, errMatch string) {
+	restore := release.MockOnClassic(false)
+	defer restore()
+
+	s.state.Lock()
+	tsk := s.state.NewTask("update-managed-boot-config", "update boot config")
+	chg := s.state.NewChange("dummy", "...")
+	chg.AddTask(tsk)
+	s.state.Unlock()
+
+	s.settle(c)
+
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	c.Assert(chg.IsReady(), Equals, true)
+	if errMatch == "" {
+		c.Check(chg.Err(), IsNil)
+		c.Check(tsk.Status(), Equals, state.DoneStatus)
+	} else {
+		c.Check(chg.Err(), ErrorMatches, errMatch)
+		c.Check(tsk.Status(), Equals, state.ErrorStatus)
+	}
+	if updateAttempted {
+		c.Assert(s.managedbl.UpdateCalls, Equals, 1)
+		if errMatch == "" && applied {
+			// we log on success
+			log := tsk.Log()
+			c.Assert(log, HasLen, 1)
+			c.Check(log[0], Matches, ".* updated boot config assets")
+			// update was applied, thus a restart was requested
+			c.Check(s.restartRequests, DeepEquals, []state.RestartType{state.RestartSystem})
+		} else {
+			// update was not applied or failed
+			c.Check(s.restartRequests, HasLen, 0)
+		}
+	} else {
+		c.Assert(s.managedbl.UpdateCalls, Equals, 0)
+	}
+}
+
+func (s *deviceMgrBootconfigSuite) TestBootConfigUpdateRunSuccess(c *C) {
```

Comment:
can these set of tests be put into a test table like

```go
for _, t := range []struct{}{
    {managedUpdated: true, updateAttempted: true, updateApplied: true},
} { 
    	s.state.Lock()
	s.setupUC20Model(c)
	s.state.Unlock()

	s.testBootConfigUpdateRun(c, t.updateAttempted, t.updateApplied, t.err)
}
```

Reply:
I'd rather stay with separate tests so that we start with a clean slate each time.